Fabric.Identity.API/scripts/Install-Identity-Utilities.psm1

```diff
@@ -1625,6 +1625,303 @@ function Test-DiscoveryService{
     }
 }
 
+function Migrate-AADSettings {
+    param (
+        [Parameter(Mandatory=$true)]
+        [string] $installConfigPath,
+        [Parameter(Mandatory=$true)]
+        [string] $azureConfigPath
+    )
+    $configSection = "identity"
+    $nodesToSearch = @("tenants", "replyUrls", "claimsIssuerTenant", "allowedTenants", "registeredApplications")
+    $childNodeGetAttribute = "name"
+    $childNodeSetAttribute = "value"
+
+    Write-DosMessage -Level "Information" -Message "Get the AAD Setting contents in install.config"
+    
+    # Quick check to see if any AAD settings are present in install.config (short circuit)
+    $existingAADSettings = Check-XMLChildNodes -installConfigPath $installConfigPath -configSection $configSection -nodeToSearch $nodesToSearch[0] -childNodeGetAttribute $childNodeGetAttribute
+    if($false -eq $existingAADSettings)
+    {
+      Write-DosMessage -Level "Information" -Message "No AAD Setting contents in install.config"
+      return
+    }
+
+    # Gather AAD Settings from install.config
+    $existingChildNodes = Get-XMLChildNodes -installConfigPath $installConfigPath -configSection $configSection -nodesToSearch $nodesToSearch -childNodeGetAttribute $childNodeGetAttribute
+
+    # Remove blank child nodes from default azuresettings.config
+    Remove-XMLChildNodes -azureConfigPath $azureConfigPath -configSection $configSection -nodesToSearch $nodesToSearch -childNodeGetAttribute $childNodeGetAttribute -childNodeAttributeSetting ""
+
+    # Add AAD settings from Doc1
+    Add-XMLChildNodesFromDoc1 -azureConfigPath $azureConfigPath -configSection $configSection -doc1ChildNodesInOrder $nodesToSearch -doc1ChildNodesToAdd $existingChildNodes
```

Comment:
Can this just be renamed to 'Add-XMLChildNodes', and change the parameters a bit as well?

Reply:
sure